src/libraries/Native/Unix/System.Security.Cryptography.Native.Android/pal_ecdsa.c

```diff
@@ -0,0 +1,114 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "pal_ecdsa.h"
+#include "pal_bignum.h"
+#include "pal_utilities.h"
+
+static jobject CryptoNative_GetEsDsaSignatureObject(JNIEnv* env)
+{
+    jstring algorithmName = JSTRING("NONEwithECDSA");
+    jobject signatureObject =
+        (*env)->CallStaticObjectMethod(env, g_SignatureClass, g_SignatureGetInstance, algorithmName);
+    (*env)->DeleteLocalRef(env, algorithmName);
+    if (CheckJNIExceptions(env))
+        return NULL;
+    return signatureObject;
+}
+
+int32_t CryptoNative_EcDsaSign(const uint8_t* dgst, int32_t dgstlen, uint8_t* sig, int32_t* siglen, EC_KEY* key)
+{
+    if (!siglen)
```

Comment:
So the rest of the arguments don't need to be checked?

Reply:
I've added some asserts for the other inputs.